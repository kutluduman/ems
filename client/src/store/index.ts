import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import auth from "./slices/auth";
import employee from "./slices/employee";

export const store = configureStore({
  reducer: {
    auth,
    employee,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;