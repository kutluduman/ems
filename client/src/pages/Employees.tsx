import { filter } from "lodash";
import { useState } from "react";
import AddIcon from "@mui/icons-material/Add";
import {
  Card,
  Table,
  Stack,
  Avatar,
  Button,
  Checkbox,
  TableRow,
  TableBody,
  TableCell,
  Container,
  Typography,
  TableContainer,
  TablePagination,
  DialogContentText,
  Snackbar,
  Alert,
} from "@mui/material";

import Page from "../components/Page";
import Scrollbar from "../components/ScrollBar";
import SearchNotFound from "../components/SearchNotFound";
import {
  EmployeesListHead,
  EmployeesListToolbar,
  EmployeesMoreMenu,
} from "../components/dashboard/employees";
import { Order } from "../components/dashboard/employees/EmployeesListHead";
import EmployeeForm from "../components/dashboard/employees/EmployeeForm";
import { useAppDispatch, useAppSelector } from "../store/hooks";
import {
  deleteEmployeeAsync,
  deleteEmployeesAsync,
  selectEmployee,
  selectEmployees,
  selectError,
  setEmployeeInQuestion,
} from "../store/slices/employee";
import Modal, { ModalAction, ModalProps } from "../components/Modal";
import { Employee } from "../models/employee";

const TABLE_HEAD = [
  { id: "name", label: "Name", alignRight: false },
  { id: "title", label: "Title", alignRight: false },
  { id: "phoneNumber", label: "Ph. Number", alignRight: false },
  { id: "address", label: "Address", alignRight: false },
  { id: "" },
];

const descendingComparator = (a: any, b: any, orderBy: any) => {
  if (b[orderBy] < a[orderBy]) {
    return -1;
  }
  if (b[orderBy] > a[orderBy]) {
    return 1;
  }
  return 0;
}

const getComparator = (order: Order, orderBy: string) => {
  return order === Order.desc
    ? (a: any, b: any) => descendingComparator(a, b, orderBy)
    : (a: any, b: any) => -descendingComparator(a, b, orderBy);
}

const applySortFilter = (array: Array<any>, comparator: any, query: any) => {
  const stabilizedThis = array.map((el: any, index: number) => [el, index]);
  stabilizedThis.sort((a: any, b: any) => {
    const order = comparator(a[0], b[0]);
    if (order !== 0) return order;
    return a[1] - b[1];
  });
  if (query) {
    return filter(
      array,
      (employee: Employee) =>
        JSON.stringify(employee).toLowerCase().indexOf(query.toLowerCase()) !==
        -1
    );
  }
  return stabilizedThis.map((el) => el[0]);
}

const Employees = () => {
  const [page, setPage] = useState(0);
  const [order, setOrder] = useState(Order.asc);
  const [selected, setSelected] = useState<Array<string>>([]);
  const [orderBy, setOrderBy] = useState("name");
  const [filterName, setFilterName] = useState("");
  const [rowsPerPage, setRowsPerPage] = useState(5);
  const [message, setMessage] = useState<string>();
  const [snackbarOpen, setSnackbarOpen] = useState<boolean>(false);
  const employees = useAppSelector(selectEmployees);
  const dispatch = useAppDispatch();
  const employee = useAppSelector(selectEmployee);
  const [openEmployeeFormModal, setOpenEmployeeFormModal] = useState(false);
  const [
    openEmployeeDeleteConfirmationModal,
    setOpenEmployeeDeleteConfirmationModal,
  ] = useState(false);
  const error = useAppSelector(selectError);

  const handleRequestSort = (event: any, property: any) => {
    const isAsc = orderBy === property && order === Order.asc;
    setOrder(isAsc ? Order.desc : Order.asc);
    setOrderBy(property);
  };

  const handleSelectAllClick = (event: any) => {
    if (event.target.checked) {
      const newSelectedItems = employees.map((n: Employee) => n._id as string);
      setSelected(newSelectedItems);
      return;
    }
    setSelected([]);
  };

  const handleClick = (event: any, _id: string) => {
    const selectedIndex = selected.indexOf(_id);
    let newSelected: Array<string> = [];
    if (selectedIndex === -1) {
      newSelected = newSelected.concat(selected, _id);
    } else if (selectedIndex === 0) {
      newSelected = newSelected.concat(selected.slice(1));
    } else if (selectedIndex === selected.length - 1) {
      newSelected = newSelected.concat(selected.slice(0, -1));
    } else if (selectedIndex > 0) {
      newSelected = newSelected.concat(
        selected.slice(0, selectedIndex),
        selected.slice(selectedIndex + 1)
      );
    }
    setSelected(newSelected);
  };

  const handleChangePage = (event: any, newPage: number) => {
    setPage(newPage);
  };

  const handleChangeRowsPerPage = (event: any) => {
    setRowsPerPage(parseInt(event.target.value, 10));
    setPage(0);
  };

  const handleFilterByName = (event: any) => {
    setFilterName(event.target.value);
  };

  const handleDeleteUser = async () => {
    setOpenEmployeeDeleteConfirmationModal(
      false
    ) as never as ModalAction["action"];
    await dispatch(deleteEmployeeAsync(employee?._id as string));
    if (error) {
      setMessage(error);
    } else {
      setMessage("Employee deleted successfully!");
    }
    setSnackbarOpen(true);
  };

  const handleDeleteMultipleUsers = async () => {
    await dispatch(deleteEmployeesAsync(selected));
    if (error) {
      setMessage(error);
    } else {
      setSelected([]);
      setMessage("Employees deleted successfully!");
    }
    setSnackbarOpen(true);
  };

  const filteredUsers = applySortFilter(
    employees,
    getComparator(order, orderBy),
    filterName
  );

  const isUserNotFound = filteredUsers.length === 0;

  return (
    <Page title="Employees">
      <Container>
        <Stack
          direction="row"
          alignItems="center"
          justifyContent="space-between"
          mb={5}
        >
          <Typography variant="h4" gutterBottom>
            Employees
          </Typography>
          <Button
            variant="contained"
            onClick={() => {
              dispatch(setEmployeeInQuestion({ employee: undefined }));
              setOpenEmployeeFormModal(true);
            }}
            startIcon={<AddIcon />}
          >
            Add New Employee
          </Button>
        </Stack>

        <Card>
          <EmployeesListToolbar
            numSelected={selected.length}
            filterName={filterName}
            onFilterName={handleFilterByName}
            handleMultipleUsersDelete={handleDeleteMultipleUsers}
          />

          <Scrollbar>
            <TableContainer sx={{ minWidth: 800 }}>
              <Table>
                <EmployeesListHead
                  order={order}
                  orderBy={orderBy}
                  headLabel={TABLE_HEAD}
                  rowCount={employees.length}
                  numSelected={selected.length}
                  onRequestSort={handleRequestSort}
                  onSelectAllClick={handleSelectAllClick}
                />
                <TableBody>
                  {filteredUsers
                    .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
                    .map((row: any) => {
                      const {
                        _id,
                        name,
                        surname,
                        phoneNumber,
                        address,
                        title,
                      } = row;
                      const isItemSelected = selected.indexOf(_id) !== -1;

                      return (
                        <TableRow
                          hover
                          key={_id}
                          tabIndex={-1}
                          role="checkbox"
                          selected={isItemSelected}
                          aria-checked={isItemSelected}
                        >
                          <TableCell padding="checkbox">
                            <Checkbox
                              checked={isItemSelected}
                              onChange={(event) => handleClick(event, _id)}
                            />
                          </TableCell>
                          <TableCell component="th" scope="row" padding="none">
                            <Stack
                              direction="row"
                              alignItems="center"
                              spacing={2}
                            >
                              <Avatar
                                alt={name}
                                src="https://minimal-kit-react.vercel.app/static/mock-images/avatars/avatar_default.jpg"
                              />
                              <Typography variant="subtitle2" noWrap>
                                {name} {surname}
                              </Typography>
                            </Stack>
                          </TableCell>
                          <TableCell align="left">{title}</TableCell>
                          <TableCell align="left">{phoneNumber}</TableCell>
                          <TableCell align="left">{address}</TableCell>
                          <TableCell align="right">
                            <EmployeesMoreMenu
                              handleEdit={() => {
                                dispatch(
                                  setEmployeeInQuestion({ employee: row })
                                );
                                setOpenEmployeeFormModal(true);
                              }}
                              handleDelete={() => {
                                dispatch(
                                  setEmployeeInQuestion({ employee: row })
                                );
                                setOpenEmployeeDeleteConfirmationModal(true);
                              }}
                            />
                          </TableCell>
                        </TableRow>
                      );
                    })}
                </TableBody>
                {isUserNotFound && (
                  <TableBody>
                    <TableRow>
                      <TableCell align="center" colSpan={6} sx={{ py: 3 }}>
                        <SearchNotFound searchQuery={filterName} />
                      </TableCell>
                    </TableRow>
                  </TableBody>
                )}
              </Table>
            </TableContainer>
          </Scrollbar>

          <TablePagination
            rowsPerPageOptions={[5, 10, 25]}
            component="div"
            count={employees.length}
            rowsPerPage={rowsPerPage}
            page={page}
            onPageChange={handleChangePage}
            onRowsPerPageChange={handleChangeRowsPerPage}
          />
        </Card>
      </Container>
      {openEmployeeFormModal && (
        <EmployeeForm
          open={openEmployeeFormModal}
          setOpen={setOpenEmployeeFormModal}
          setMessage={setMessage}
          setSnackbarOpen={setSnackbarOpen}
        />
      )}
      <Modal
        title="Are you sure?"
        actions={[
          {
            title: "No",
            action: () =>
              setOpenEmployeeDeleteConfirmationModal(
                false
              ) as never as ModalAction["action"],
          },
          {
            title: "Yes",
            action: handleDeleteUser,
          },
        ]}
        open={openEmployeeDeleteConfirmationModal}
        setOpen={
          setOpenEmployeeDeleteConfirmationModal as never as ModalProps["setOpen"]
        }
      >
        <DialogContentText id="alert-dialog-description">
          Are you sure you want to delete {employee?.name}?
        </DialogContentText>
      </Modal>
      {snackbarOpen && (
        <Snackbar
          open={snackbarOpen}
          autoHideDuration={6000}
          message={message}
          onClose={() => setSnackbarOpen(false)}
        >
          <Alert severity={error ? "error" : "success"} sx={{ width: "100%" }}>
            {message}
          </Alert>
        </Snackbar>
      )}
    </Page>
  );
}

export default Employees;
