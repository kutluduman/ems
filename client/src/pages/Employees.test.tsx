import { create } from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";

import { store } from "../store";
import Employees from "./Employees";

describe("Employees Component", () => {
  test("renders the App Component", () => {
    const tree = create(
      <Provider store={store}>
        <Router>
          <Employees />
        </Router>
      </Provider>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});