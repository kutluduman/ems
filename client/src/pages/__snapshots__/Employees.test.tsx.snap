// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employees Component renders the App Component 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
>
  <div
    className="css-d3ri6l-MuiStack-root"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
    >
      All Employees
    </h4>
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </span>
      Add New Employee
    </button>
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-obvsh9-MuiToolbar-root"
    >
      <div
        className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedStart css-1dw4ior-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <div
          className="MuiInputAdornment-root MuiInputAdornment-positionStart css-ittuaa-MuiInputAdornment-root"
        >
          <span
            className="notranslate"
          >
            ​
          </span>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiBox-root css-1qhky1k-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
        <input
          className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Search employee..."
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-nnbavb"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      className="css-jjtu05"
    >
      <div
        className="css-wkgdtb"
        data-simplebar={true}
      >
        <div
          className="simplebar-wrapper"
        >
          <div
            className="simplebar-height-auto-observer-wrapper"
          >
            <div
              className="simplebar-height-auto-observer"
            />
          </div>
          <div
            className="simplebar-mask"
          >
            <div
              className="simplebar-offset"
            >
              <div
                className="simplebar-content-wrapper"
              >
                <div
                  className="simplebar-content"
                >
                  <div
                    className="MuiTableContainer-root css-dac6fd-MuiTableContainer-root"
                  >
                    <table
                      className="MuiTable-root css-rqglhn-MuiTable-root"
                      role={null}
                    >
                      <thead
                        className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
                        role={null}
                      >
                        <tr
                          className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                          role={null}
                        >
                          <th
                            aria-sort={null}
                            className="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingCheckbox MuiTableCell-sizeMedium css-hbtnrh-MuiTableCell-root"
                            scope="col"
                          >
                            <span
                              className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                              onBlur={[Function]}
                              onContextMenu={[Function]}
                              onDragLeave={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchMove={[Function]}
                              onTouchStart={[Function]}
                              tabIndex={null}
                            >
                              <input
                                checked={false}
                                className="PrivateSwitchBase-input css-1m9pwf3"
                                data-indeterminate={false}
                                onChange={[Function]}
                                type="checkbox"
                              />
                              <svg
                                aria-hidden={true}
                                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                                data-testid="CheckBoxOutlineBlankIcon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                                />
                              </svg>
                            </span>
                          </th>
                          <th
                            aria-sort="ascending"
                            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                            scope="col"
                          >
                            <span
                              className="MuiButtonBase-root MuiTableSortLabel-root Mui-active css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                              onBlur={[Function]}
                              onClick={[Function]}
                              onContextMenu={[Function]}
                              onDragLeave={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchMove={[Function]}
                              onTouchStart={[Function]}
                              role="button"
                              tabIndex={0}
                            >
                              Name
                              <div
                                className="MuiBox-root css-trfuzi"
                              >
                                sorted ascending
                              </div>
                              <svg
                                aria-hidden={true}
                                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                                data-testid="ArrowDownwardIcon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                                />
                              </svg>
                            </span>
                          </th>
                          <th
                            aria-sort={null}
                            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                            scope="col"
                          >
                            <span
                              className="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                              onBlur={[Function]}
                              onClick={[Function]}
                              onContextMenu={[Function]}
                              onDragLeave={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchMove={[Function]}
                              onTouchStart={[Function]}
                              role="button"
                              tabIndex={0}
                            >
                              Title
                            </span>
                          </th>
                          <th
                            aria-sort={null}
                            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                            scope="col"
                          >
                            <span
                              className="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                              onBlur={[Function]}
                              onClick={[Function]}
                              onContextMenu={[Function]}
                              onDragLeave={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchMove={[Function]}
                              onTouchStart={[Function]}
                              role="button"
                              tabIndex={0}
                            >
                              Number
                            </span>
                          </th>
                          <th
                            aria-sort={null}
                            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                            scope="col"
                          >
                            <span
                              className="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                              onBlur={[Function]}
                              onClick={[Function]}
                              onContextMenu={[Function]}
                              onDragLeave={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchMove={[Function]}
                              onTouchStart={[Function]}
                              role="button"
                              tabIndex={0}
                            >
                              Address
                            </span>
                          </th>
                          <th
                            aria-sort={null}
                            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                            scope="col"
                          >
                            <span
                              className="MuiButtonBase-root MuiTableSortLabel-root css-1qgma8u-MuiButtonBase-root-MuiTableSortLabel-root"
                              onBlur={[Function]}
                              onClick={[Function]}
                              onContextMenu={[Function]}
                              onDragLeave={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchMove={[Function]}
                              onTouchStart={[Function]}
                              role="button"
                              tabIndex={0}
                            />
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                        role={null}
                      />
                      <tbody
                        className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                        role={null}
                      >
                        <tr
                          className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
                          role={null}
                        >
                          <td
                            aria-sort={null}
                            className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter MuiTableCell-sizeMedium css-ef9479-MuiTableCell-root"
                            colSpan={6}
                          >
                            <div
                              className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
                            >
                              <h6
                                className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-alignCenter MuiTypography-gutterBottom css-kxw4um-MuiTypography-root"
                              >
                                Not found
                              </h6>
                              <p
                                className="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1ypfif-MuiTypography-root"
                              >
                                No results found for  
                                <strong>
                                  "
                                  
                                  "
                                </strong>
                                . Try checking for typos or using complete words.
                              </p>
                            </div>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="simplebar-placeholder"
          />
        </div>
        <div
          className="simplebar-track simplebar-horizontal"
        >
          <div
            className="simplebar-scrollbar"
          />
        </div>
        <div
          className="simplebar-track simplebar-vertical"
        >
          <div
            className="simplebar-scrollbar"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiTablePagination-root css-jtlhu6-MuiTablePagination-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-11bfvty-MuiToolbar-root-MuiTablePagination-toolbar"
      >
        <div
          className="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
        />
        <p
          className="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
        >
          Rows per page:
        </p>
        <div
          className="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
          onClick={[Function]}
          variant="standard"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            className="MuiTablePagination-select MuiSelect-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            5
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            onAnimationStart={[Function]}
            onChange={[Function]}
            tabIndex={-1}
            value={5}
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTablePagination-selectIcon MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
        <p
          className="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
        >
          0–0 of 0
        </p>
        <div
          className="MuiTablePagination-actions"
        >
          <button
            aria-label="Go to previous page"
            className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            title="Go to previous page"
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowLeftIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
              />
            </svg>
          </button>
          <button
            aria-label="Go to next page"
            className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            title="Go to next page"
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
